rest/src/main/kotlin/json/response/DiscordErrorResponse.kt

```diff
@@ -1,33 +1,73 @@
 package dev.kord.rest.json.response
 
 import dev.kord.rest.json.JsonErrorCode
+import kotlinx.serialization.KSerializer
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonObject
 
 /**
  * Represents a [Discord error response](https://discord.com/developers/docs/topics/opcodes-and-status-codes#json).
  */
 @Serializable
 class DiscordErrorResponse(
-        val code: JsonErrorCode = JsonErrorCode.Unknown,
-        val errors: Map<String, DiscordFieldError> = emptyMap(),
-        val message: String = "",
+    val code: JsonErrorCode = JsonErrorCode.Unknown,
+    val errors: Map<String, DiscordFieldError> = emptyMap(),
+    val message: String = "",
 )
 
 /**
  * An error for a specific field.
  */
-@Serializable
+@Serializable(DiscordFieldError.Serializer::class)
 class DiscordFieldError(
-        @SerialName("_errors")
-        val errors: List<DiscordErrorDetail>,
-)
+    @SerialName("_errors")
+    val errors: List<DiscordErrorDetail> = emptyList(),
+    val optionErrors: List<Map<String, DiscordFieldError>> = emptyList()
```

Comment:
I don't think there's a field consistently named `options`.
What I think is going on is that Discord provides the name of the failed property as a key, but I'm not certain, did you check this on multiple invalid structures?

Reply:
this is consistently named options for slash commands with options else it will contain a `_errors` field yeah.
I don't assume it to be named `options` I go through the elements looking for what we know exists which is `_errors`